### PR TITLE
cassandra reaper

### PR DESCRIPTION
**Pull request by developer-guy**
### Image Size

- [x] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes: size is reduced from `482MB` to `309MB`.

### Image Vulnerabilities

- [] The Grype vulnerability scan returned 0 CVE(s).
- [x] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes: We created necessary advisory entries for cassandra-reaper CVEs

### Image Tagging

- [ ] The image is _not_ tagged with version tags.
- [x] The image is tagged with :latest
- [ ] The image is not tagged with :latest (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [x] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes:

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [ ] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [ ] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [x] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [ ] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [ ] Environment variables added.
- [ ] Environment variables not added and not required.

### SIGTERM

- [ ] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [ ] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

File: cassandra-reaper [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit da5096f71cd25558bd60e26e830c29e78d0e1160
```
Comment (imjasonh) [inline cassandra-reaper:1]: Is this intentional?

Comment (developer-guy) [reply cassandra-reaper:1]: nope, its my bad 🤦

**Comment by imjasonh** [comment]
```
  Test failed for ref registry.local:5000/testing/cassandra-reaper@sha256:a30f048012a959323c29097add6e66326501632723d59a463904101c5aaaa477, got error: exit status 1
  + write a trap function for cleanup helm uninstall
  Try 'write --help' for more information.

```

**Comment by developer-guy** [comment]
> ```
>   Test failed for ref registry.local:5000/testing/cassandra-reaper@sha256:a30f048012a959323c29097add6e66326501632723d59a463904101c5aaaa477, got error: exit status 1
>   + write a trap function for cleanup helm uninstall
>   Try 'write --help' for more information.
> ```

I missed the comment line, sorry, fixed

File: images/cassandra-reaper/tests/smoke-test.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# list all the storageclasses
+
+kubectl get storageclass
+
+# we have to install cert-manager first
+
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+helm install cert-manager jetstack/cert-manager \
+     --namespace cert-manager --create-namespace --set installCRDs=true
```
Comment (imjasonh) [inline images/cassandra-reaper/tests/smoke-test.sh:21]: Hard-coding this means we won't be able to run two of these tests side-by-side. Can you make this namespace `cert-manager-${RANDOM}` to help deconflict?

Comment (developer-guy) [reply images/cassandra-reaper/tests/smoke-test.sh:21]: I added RANDOM to every release name and namespace

File: images/cassandra-reaper/tests/smoke-test.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+# list all the storageclasses
+
+kubectl get storageclass
+
+# we have to install cert-manager first
+
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+helm install cert-manager-${RANDOM} jetstack/cert-manager \
+     --namespace cert-manager-${RANDOM} --create-namespace --set installCRDs=true
```
Comment (imjasonh) [inline images/cassandra-reaper/tests/smoke-test.sh:21]: Each time you get `${RANDOM}` it'll give you a different value, so this won't work.

You should do something like

```
name=cert-manager-${RANDOM}
namespace=cert-manager-${RANDOM}

helm install $name jetstack/cert-manager \
     --namespace $namespace --create-namespace --set installCRDs=true

kubectl wait --for=condition=ready pod --selector app.kubernetes.io/instance=cert-manager --namespace $namespace
...
```

Comment (developer-guy) [reply images/cassandra-reaper/tests/smoke-test.sh:21]: why? I created an environment variable called $RANDOM in the tests through a terraform resource random_pet ? 🤔

Comment (developer-guy) [reply images/cassandra-reaper/tests/smoke-test.sh:21]: done

**Review by dlorenc — APPROVED** [review]

**Pull request merged**